Sources/SwiftDocC/Model/Rendering/Variants/JSONPointer.swift

```diff
@@ -87,36 +87,78 @@ public struct JSONPointer: Codable, CustomStringConvertible, Equatable {
     public init(from decoder: Decoder) throws {
         let container = try decoder.singleValueContainer()
         let stringValue = try container.decode(String.self)
-        self.pathComponents = stringValue.removingLeadingSlash.components(separatedBy: "/").map(Self.unescape)
+        self.pathComponents = Self.unescaped(stringValue)
     }
     
-    /// Escapes a path component of a JSON pointer.
-    static func escape(_ pointerPathComponents: String) -> String {
-        applyEscaping(pointerPathComponents, shouldUnescape: false)
-    }
-    
-    /// Unescaped a path component of a JSON pointer.
-    static func unescape(_ pointerPathComponents: String) -> String {
-        applyEscaping(pointerPathComponents, shouldUnescape: true)
+    private static func escaped(_ pathComponents: [String]) -> String {
+        // This code is called quite frequently for mixed language content.
+        // Optimizing it has a measurable impact on the total documentation build time.
+        
+        var string: [UTF8.CodeUnit] = []
+        string.reserveCapacity(
+            pathComponents.reduce(0) { acc, component in
```

Comment:
It shouldn't be and I couldn't measure any difference. 

`reduce(_:)` is [an accumulating for-loop](https://github.com/swiftlang/swift/blob/main/stdlib/public/core/SequenceAlgorithms.swift#L666-L676) and the compiler shouldn't have any problems inlining and optimizing it the same as I wrote the loop inline myself.

Reply:
> reduce(_:) is [an accumulating for-loop](https://github.com/swiftlang/swift/blob/main/stdlib/public/core/SequenceAlgorithms.swift#L666-L676) and the compiler shouldn't have any problems inlining and optimizing it the same as if I wrote the loop inline myself.

Note that this is only true because the accumulating value is a primitive (`Int`). 

Using `reduce(_:)` over collections result in repeated copies that makes it accidentally _O(n²)_ instead of _O(n)_. Using `reduce(into:)` instead for accumulating into collections solves this.